temporalio/client.py

```diff
@@ -5428,6 +5995,165 @@ async def start_workflow_update(
+            namespace=self._client.namespace,
+            workflow_execution=temporalio.api.common.v1.WorkflowExecution(
+                workflow_id=workflow_id,
+                run_id=run_id or "",
+            ),
+            request=temporalio.api.update.v1.Request(
+                meta=temporalio.api.update.v1.Meta(
+                    update_id=input.update_id or str(uuid.uuid4()),
+                    identity=self._client.identity,
+                ),
+                input=temporalio.api.update.v1.Input(
+                    name=input.update,
+                ),
+            ),
+            wait_policy=temporalio.api.update.v1.WaitPolicy(
+                lifecycle_stage=temporalio.api.enums.v1.UpdateWorkflowExecutionLifecycleStage.ValueType(
+                    input.wait_for_stage
+                )
+            ),
+        )
+        if input.args:
+            req.request.input.args.payloads.extend(
+                await self._client.data_converter.encode(input.args)
+            )
+        if input.headers is not None:
+            temporalio.common._apply_headers(
+                input.headers, req.request.input.header.fields
+            )
+        return req
+
+    async def start_update_with_start_workflow(
+        self, input: StartWorkflowUpdateWithStartInput
+    ) -> WorkflowUpdateHandle[Any]:
+        seen_start = False
+
+        def on_start(
+            start_response: temporalio.api.workflowservice.v1.StartWorkflowExecutionResponse,
+        ):
+            nonlocal seen_start
+            if not seen_start:
+                input._on_start(start_response)
+                seen_start = True
+
+        err: BaseException
+
+        try:
+            return await self._start_workflow_update_with_start(
+                input.start_workflow_input, input.update_workflow_input, on_start
+            )
+        except asyncio.CancelledError as _err:
+            err = _err
+            raise WorkflowUpdateRPCTimeoutOrCancelledError() from err
+        except RPCError as _err:
+            err = _err
+            if err.status in [
+                RPCStatusCode.DEADLINE_EXCEEDED,
+                RPCStatusCode.CANCELLED,
+            ]:
+                raise WorkflowUpdateRPCTimeoutOrCancelledError() from err
+            else:
+                multiop_failure = (
+                    temporalio.api.errordetails.v1.MultiOperationExecutionFailure()
+                )
+                if err.grpc_status.details[0].Unpack(multiop_failure):
+                    status = next(
+                        (
+                            st
+                            for st in multiop_failure.statuses
+                            if (
+                                st.details
+                                and not st.details[0].Is(
+                                    temporalio.api.failure.v1.MultiOperationExecutionAborted.DESCRIPTOR
+                                )
+                            )
```

Comment:
What did we end up deciding here about what a successful start but failed update looks like? I think server side today `st.details` is never `None` correct? Should we ignore `OK` statuses? This is super rare and so non-blocking for this PR, but whatever the decision we may need to apply to other SDKs too.

Reply:
I'm skipping `OK` statuses in Python and TS.